docs/blog/_posts/2018-03-05-seventh-dotty-milestone-release.md

```diff
@@ -0,0 +1,259 @@
+List(1, 2, 3).flatten             // error: Cannot prove that Int <:< List[Y]
+```
+
+The implicit evidence `ev` is only used to constrain the type parameter `X` of `List` such that we
+can safely cast from `X` to `List[_]`. The usage of the `erased` modifier ensures that the evidence
+is not used and can be safely removed at compilation time.
+
+For more information, visit the [Erased Terms](http://dotty.epfl.ch/docs/reference/erased-terms.html)
+section of our documentation.
+
+**Note**: Erased terms replace _phantom types_: they have similar semantics, but with the added
+advantage that any type can be an erased parameter. See [#3410](https://github.com/lampepfl/dotty/pull/3410).
+
+
+### Improved IDE support [#3960](https://github.com/lampepfl/dotty/pull/3960)
+The Dotty language server now supports context sensitive IDE completions. Completions now include
+local and imported definitions. Members completions take possible implicit conversions into account.
+
+<!-- A GIF would be nice? -->
+
+We also improved the `find references` functionality. It is more robust and much faster!
+
+Try it out in [Visual Studio Code](http://dotty.epfl.ch/docs/usage/ide-support.html)!
+
+### Improvements in GADT type inference
+
+GADT typechecking is an advanced feature that got significantly improved in this
+release. GADTs are case class hierarchies similar to this one:
+
+```scala
+sealed trait Exp[T]
+case class IntLit(n: Int) extends Exp[Int]
+case class BooleanLit(b: Boolean) extends Exp[Boolean]
+
+case class GenLit[T](t: T) extends Exp[T]
+case class Plus(e1: Exp[Int], e2: Exp[Int]) extends Exp[Int]
+case class Fun[S, T](f: Exp[S] => Exp[T]) extends Exp[S => T]
+case class App[T, U](f: Exp[T => U], e: Exp[T]) extends Exp[U]
+```
+
+where different constructors, such as `IntLit` and `BooleanLit`, pass different type argument to the super trait. Hence, typechecking a pattern match on `v: Exp[T]` requires special care: for instance, if `v = IntLit(5)` then `T` must be `Int`. This enables writing a typed interpreter `eval[T](e: Exp[T]): T`. In each pattern matching branch
+
+```scala
+object Interpreter {
+  def eval[T](e: Exp[T]): T = e match {
+    case IntLit(n) => // Here T = Int and n: Int
+      n
+    case BooleanLit(b) => // Similarly, here T = Boolean and b: Boolean
+      b
+
+    case GenLit(t) => //Here t: T
+
+      // the next line is an error, but was allowed before the fix to https://github.com/lampepfl/dotty/issues/1754:
+      //val w: GenLit[Nothing] = w
+
+      t
+
+    case Plus(e1, e2) =>
+      // Here T = Int and e1, e2: Exp[Int]
+      eval(e1) + eval(e2)
+
+    // The next cases triggered warnings before the fix to
+    // https://github.com/lampepfl/dotty/issues/3666
+
+    case f: Fun[s, t]  => // Here T = s => t
+      (v: s) => eval(f.f(GenLit(v)))
+
+    case App(f, e)     => // Here f: Exp[s, T] and e: Exp[s]
+      eval(f)(eval(e))
+  }
+}
+```
+
+Earlier Dotty releases had issues typechecking such interpreters.
```

Comment:
I think it's important to note that Scala 2 also has unresolved issues typechecking GADTs, like with the neg example in https://github.com/lampepfl/dotty/pull/3990/files#diff-e7b3b26b928fa129081481ffc513af95

Reply:
Added a bit; I think Dotty's already much better than Scalac, but I haven't tried enough personally to make strong claims this in release notes. Feel free to add stronger claims you feel comfortable with.